### ISSUE TITLE
WithMaxDuration seems broken because StartTime is zero

### ISSUE THREAD

**Issue by srenatus** [failsafe-go/failsafe-go#29]
Hey there! Thanks for this nice an comprehensive package.

Using v0.4.4, I've stumbled upon this bug:

```go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/failsafe-go/failsafe-go"
	"github.com/failsafe-go/failsafe-go/retrypolicy"
)

type result any

var mehErr = fmt.Errorf("meh")

func retrier() (result, error) {
	policy := retrypolicy.Builder[result]().
		HandleErrors(mehErr).
		WithDelay(time.Millisecond).
		WithMaxDuration(3 * 24 * time.Hour).
		WithMaxAttempts(2).
		OnFailure(func(ee failsafe.ExecutionEvent[result]) {
			log.Printf("start time: %v", ee.StartTime())
			log.Printf("elapsed time: %v", ee.ElapsedTime())
		}).
		Build()

	return failsafe.Get(func() (result, error) {
		return nil, mehErr
	}, policy)
}

func main() {
	log.Println(retrier())
}
```

This code works as expected **if** `WithMaxDuration` is commented out. When it's like above, it'll fail immediately, because the unset StartTime makes ElapsedTime be extremely high:

```
2009/11/10 23:00:00 start time: 0001-01-01 00:00:00 +0000 UTC
2009/11/10 23:00:00 elapsed time: 2562047h47m16.854775807s
2009/11/10 23:00:00 <nil> retries exceeded. last result: <nil>, last error: meh
```

[See this playground](https://play.golang.com/p/CwSKrrhQpvq)

Happy to put up a PR, but I wasn't about where to best put the `StartTime = time.Now()`.

**Comment by jhalterman** [comment]
This has been fixed in v0.4.5.

**Comment by srenatus** [comment]
Thank you!

**Comment by jhalterman** [comment]
Also released this in v0.5.0, which includes the new [hedge policy](https://failsafe-go.dev/hedge/).
